src/o365/graph/commands/teams/teams-set.ts

```diff
@@ -0,0 +1,171 @@
+}
+
+class GraphTeamsSetCommand extends GraphCommand {
+  private static props: string[] = [
+    'displayName',
+    'description',
+    'mailNickName',
+    'classification',
+    'visibility '
+  ];
+
+  public get name(): string {
+    return `${commands.TEAMS_SET}`;
+  }
+
+  public get description(): string {
+    return 'Updates settings of a Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    GraphTeamsSetCommand.props.forEach((p: string) => {
+        telemetryProps[p] = typeof (args.options as any)[p] !== 'undefined';
+    });
+    return telemetryProps;
+  }
+
+  private mapRequestBody(options: Options): any {
+    const requestBody: any = {};
+    if (options.displayName) {
+      requestBody.displayName = options.displayName;
+    }
+    if (options.description) {
+      requestBody.description = options.description;
+    }
+    if (options.mailNickName) {
+      requestBody.mailNickName = options.mailNickName;
+    }
+    if (options.classification) {
+      requestBody.classification = options.classification;
+    }
+    if (options.visibility) {
+      requestBody.visibility = options.visibility;
+    }
+    return requestBody;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): Promise<{}> => {
+        const body: any = this.mapRequestBody(args.options);
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/beta/groups/${encodeURIComponent(args.options.teamId)}`,
+          headers: {
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata.metadata=none'
+          },
+          body: body,
+          json: true
+        };
+
+        return request.patch(requestOptions);
+      })
+      .then((): void => {
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --teamId <teamId>',
+        description: 'The ID of the Microsoft Teams team for which to update settings'
+      },
+      {
+        option: '--displayName [displayName]',
+        description: 'The display name for the Microsoft Teams team'
+      },
+      {
+        option: '--description [description]',
+        description: 'The description for the Microsoft Teams team'
+      },
+      {
+        option: '--mailNickName [mailNickName]',
+        description: 'The mail alias for the Microsoft Teams team'
+      },
+      {
+        option: '--classification [classification]',
+        description: 'The classification for the Microsoft Teams team with valid values HBI, MBI, LBI, GDPR'
```

Comment:
Aren't the classification values configurable in the AAD rather than being fixed values?

Reply:
Absolutely right, Should remove the hint